cec.cpp

```diff
@@ -640,7 +675,7 @@ int log_cb(void * self, const cec_log_message message) {
          time,
          msg);
    trigger_event(EVENT_LOG, args);
```

Comment:
If `Py_BuildValue` has failed here and `args` is `NULL`, it doesn't seem like this should call `trigger_event`. I think a better way to prevent the crash here is to explicitly check for `NULL` and avoid both the call to `trigget_event` and `Py_DECREF`.

It would probably be useful to have an error message in this situation as well, since `Py_BuildValue` is called with simple arguments, and a failure of `Py_BuildValue` is probably a bug.

Reply:
Sure, I was just focusing on preventing a crash, because it actually happened to me. I'm not sure what caused it though.

Reply:
So the failure of `Py_BuildValue` was caused by a log message containing invalid characters:

`Recorder 1 (1): menu language set to 'eng\333\333\333'`

I've fixed that in commit 29d4069.

Reply:
Thanks!